### PR TITLE
feat: refactor infra yaml struct

### PR DESCRIPTION
**Pull request by taorzhang**
[SKIP CI]sealos:  refactor infra yaml struct

### CODE REVIEW

File: pkg/infra/README.md [hunk 1]
```diff
@@ -3,68 +3,68 @@
...
+    - roles: [master, aaa, bbb] # required
+      count: 3 # Required
+      cpu: 2
       memory: 4
-      arch: amd64 # ENUM: amd64/arm64 (NOTE: the default value is amd64)
-	  ecsType: "ecs.n1.medium" #fixed ecs type
-      os: # 默认 CentOS 7.6
-        name: CentOS # ENUM: CentOS/Ubuntu/Debain and so on.
-        version: 7.6
-        id: xxx # 允许用户指定已有的镜像ID （Optional）
```
Comment (cuisongliu) [inline pkg/infra/README.md:37]: ID Is it more appropriate to keep it? Because sometimes users directly fix the OS ID.

File: pkg/infra/README.md [hunk 2]
```diff
@@ -3,68 +3,68 @@
...
-        sealos.io/VpcID: v-xxxxx
-        sealos.io/VSwitchID: v-xxxxx
-        sealos.io/SecurityGroupID: v-xxxxx
-    accessChannels: # If all access channels are the same, it will be filled.
+    zoneIds: [cn-hangzhou-a, cn-hangzhou-b]
+    accessChannels:
       ssh:
-        passwd: xxx #If the passwd is empty, a random password will be generated.
+        username: ubuntu
+        # ENUM: passfile/pkfile
```
Comment (cuisongliu) [inline pkg/infra/README.md:29]: Is the structure of direct reuse cluster.SSH a little bad?
```
type SSH struct {
	User     string `json:"user,omitempty"`
	Passwd   string `json:"passwd,omitempty"`
	Pk       string `json:"pk,omitempty"`
	PkPasswd string `json:"pkPasswd,omitempty"`
	Port     uint16 `json:"port,omitempty"`
}
```

File: pkg/infra/README.md [hunk 3]
```diff
@@ -3,68 +3,68 @@
...
+    sealos.io/SubnetID: v-xxxxx
+    sealos.io/SecurityGroupID: v-xxxxx
+    sealos.io/EIPID: v-xxxxx
 spec:
   provider: AliyunProvider
   credential:
     accessKey: xxx
     accessSecret: xxx
   cluster:
-    isSeize: true
```
Comment (cuisongliu) [inline pkg/infra/README.md:16]: How to support spot instances

Comment (fanux) [reply pkg/infra/README.md:16]: annotations

File: pkg/infra/README.md (hunk 2 above)
Comment (fanux) [reply pkg/infra/README.md:29]: agree

**Comment by fanux** [comment]
```
apiVersion: xxx
kind: Infra
metadata:
  name: aws-infra-demo
  annotations:
    sealos.io/VpcID: v-xxxxx
    sealos.io/RouteID: v-xxxxx
    sealos.io/IgwID: v-xxxxxx
    sealos.io/EgwID: v-xxxxxx
    sealos.io/SubnetID: v-xxxxx
    sealos.io/SecurityGroupID: v-xxxxx
    sealos.io/EIPID: v-xxxxx
spec:
    regionIds: [cn-hangzhou, cn-shanghai]
    zoneIds: [cn-hangzhou-a, cn-hangzhou-b]
    ssh:
       passwd: xxx
       pk: /root/.ssh/id_rsa
       port: 22
       user: root
   hosts:
   - roles: [master, aaa, bbb] # required
     count: 3 # Required
     cpu: 2
     memory: 4
     # ENUM: amd64/arm64 (NOTE: the default value is amd64)
     arch: amd64
     # ENUM: ubuntu:20.04, centos:7.2 and so on.
     image: utuntu:20.04
     disks:
     - capacity: 50
          # ENUM: system/data
          type: system
```

File: pkg/infra/README.md [hunk 4]
```diff
@@ -3,68 +3,65 @@
...
-        version: 7.6
-        id: xxx # 允许用户指定已有的镜像ID （Optional）
+    - roles: [master, aaa, bbb] # required
+      count: 3 # Required
+      resources:
+        cpu: 2
+        memory: 4
+      # ENUM: amd64/arm64 (NOTE: the default value is amd64)
+      arch: amd64
+      image: ubuntu:20.04
```
Comment (cuisongliu) [inline pkg/infra/README.md:40]: Does this place refer to the mirror ID?

File: pkg/infra/README.md [hunk 5]
```diff
@@ -3,68 +3,65 @@
...
-  name: alicloud-infra-demo
+  name: aws-infra-demo
+  annotations:
+    sealos.io/VpcID: v-xxxxx
+    sealos.io/RouteID: v-xxxxx
+    sealos.io/IgwID: v-xxxxxx
+    sealos.io/EgwID: v-xxxxxx
+    sealos.io/SubnetID: v-xxxxx
+    sealos.io/SecurityGroupID: v-xxxxx
+    sealos.io/EIPID: v-xxxxx
```
Comment (cuisongliu) [inline pkg/infra/README.md:17]: What kind of field should I add if it is spot instances?

**Review by cuisongliu — APPROVED** [review]

**Pull request merged**